codegen/generator/src/Visitors/MetadataQueryParamVisitor.cs

```diff
@@ -0,0 +1,130 @@
+            method.Signature.Parameters.Any(p => p.Type.IsDictionary && p.Name == "metadata"))
+        {
+            ValueExpression? uri = null;
+            var statements = method.BodyStatements?.ToList() ?? new List<MethodBodyStatement>();
+            VisitExplodedMethodBodyStatements(
+                statements!,
+                statement =>
+                {
+                    // Check if the statement is an assignment to a variable named "uri"
+                    // Capture it if so
+                    if (statement is ExpressionStatement expressionStatement &&
+                       expressionStatement.Expression is AssignmentExpression assignmentExpression &&
+                       assignmentExpression.Variable is DeclarationExpression declarationExpression &&
+                       declarationExpression.Variable is VariableExpression variableExpression &&
+                       variableExpression.Declaration.RequestedName == "uri")
+                    {
+                        uri = variableExpression;
+                    }
+                    // Try to remove the unnecessary list declaration
+                    if (statement is ExpressionStatement expressionStatement2 &&
+                       expressionStatement2.Expression is AssignmentExpression assignmentExpression2 &&
+                       assignmentExpression2.Variable is DeclarationExpression declarationExpression2 &&
+                       declarationExpression2.Variable is VariableExpression variableExpression2 &&
+                       variableExpression2.Declaration.RequestedName == "list" &&
+                       variableExpression2.Type.IsCollection && variableExpression2.Type.IsGenericType)
+                    {
+                        // Remove the list declaration
+                        return new SingleLineCommentStatement("Plugin customization: remove unnecessary list declaration");
+                    }
+
+                    if (uri is not null &&
+                        statement is ForEachStatement foreachStatement &&
+                        foreachStatement.Enumerable is DictionaryExpression dictionaryExpression &&
+                        dictionaryExpression.Original is VariableExpression variable &&
+                        variable.Declaration.RequestedName == "metadata")
+                    {
+                        var formatString = new FormattableStringExpression("metadata[{0}]", [foreachStatement.ItemVariable.Property("Key")]);
+                        var appendQueryStatement = uri.Invoke("AppendQuery", [formatString, foreachStatement.ItemVariable.Property("Value"), new KeywordExpression("true", null)]);
+                        foreachStatement.Body.Clear();
+                        foreachStatement.Body.Add(new SingleLineCommentStatement("Plugin customization: Properly handle metadata query parameters"));
+                        foreachStatement.Body.Add(new ExpressionStatement(appendQueryStatement));
```

Comment:
This can simply be
```suggestion
                        foreachStatement.Body.Add(appendQueryStatement.Terminate());
```
which appends a semicolon and does the same thing under the hood.